source/ida_interface.cc

```diff
@@ -152,6 +152,9 @@ IDAInterface<VEC>::~IDAInterface()
 {
   if (ida_mem)
     IDAFree(&ida_mem);
+#ifdef DEAL_II_WITH_MPI
+  MPI_Comm_free(&communicator);
```

Comment:
why do we need this here? It should be the role of the owner of the communicator to free it ...